cdap-security/src/main/java/io/cdap/cdap/security/authorization/AllowlistAccessEnforcer.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright © 2023 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.security.authorization;
+
+import com.google.inject.Inject;
+import io.cdap.cdap.api.security.AccessException;
+import io.cdap.cdap.common.conf.CConfiguration;
+import io.cdap.cdap.common.internal.remote.RemoteClientFactory;
+import io.cdap.cdap.proto.element.EntityType;
+import io.cdap.cdap.proto.id.EntityId;
+import io.cdap.cdap.proto.security.Permission;
+import io.cdap.cdap.proto.security.Principal;
+import java.util.Set;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Implementation of the RemoteAccessEnforcer. It is used to skip authorization enforcement for
+ * some users.
+ */
+public class AllowlistAccessEnforcer extends RemoteAccessEnforcer {
```

Comment:
Can we change this so that we wrap the RemoteAccessEnforcer as a delegate instead of extending it? i.e. bind the RemoteAccessEnforcer as a named binding in the module, and inject it as the delegate. That will be more extensible and less prone to errors in the long run. Thanks!

Reply:
ack